src/android/nl/xservices/plugins/HeadsetDetection.java

```diff
@@ -28,36 +29,57 @@
   }
 
   @Override
   public void initialize(CordovaInterface cordova, CordovaWebView webView) {
-      super.initialize(cordova, webView);
-      mCachedWebView = webView;
-      IntentFilter intentFilter = new IntentFilter();
-      intentFilter.addAction(Intent.ACTION_HEADSET_PLUG);
-      this.receiver = new BroadcastReceiver() {
-          @Override
-          public void onReceive(Context context, Intent intent) {
-              if (intent.getAction().equals(Intent.ACTION_HEADSET_PLUG)) {
-                  int state = intent.getIntExtra("state", -1);
-                  switch (state) {
-                  case 0:
-                      Log.d(LOG_TAG, "Headset is unplugged");
-                      mCachedWebView.sendJavascript("cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetRemoved();");
-                      break;
-                  case 1:
-                      Log.d(LOG_TAG, "Headset is plugged");
-                      mCachedWebView.sendJavascript("cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetAdded();");
-                      break;
-                  default:
-                      Log.d(LOG_TAG, "I have no idea what the headset state is");
-                  }
-              }
+    super.initialize(cordova, webView);
+    mCachedWebView = webView;
+    IntentFilter intentFilter = new IntentFilter();
+    intentFilter.addAction(Intent.ACTION_HEADSET_PLUG);
+    intentFilter.addAction(BluetoothHeadset.ACTION_CONNECTION_STATE_CHANGED);
+    this.receiver = new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        int state = -1;
+        if (intent.getAction().equals(Intent.ACTION_HEADSET_PLUG)) {
+          state = intent.getIntExtra("state", -1);
+          switch (state) {
+          case 0:
+            Log.d(LOG_TAG, "Headset is unplugged");
+            mCachedWebView.sendJavascript(
+                "cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetRemoved();");
+            break;
+          case 1:
+            Log.d(LOG_TAG, "Headset is plugged");
+            mCachedWebView.sendJavascript(
+                "cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetAdded();");
+            break;
+          default:
+            Log.d(LOG_TAG, "I have no idea what the headset state is");
           }
-      };
-      mCachedWebView.getContext().registerReceiver(this.receiver, intentFilter);
+        } else if (intent.getAction().equals(BluetoothHeadset.ACTION_CONNECTION_STATE_CHANGED)) {
+          state = intent.getIntExtra(BluetoothHeadset.EXTRA_STATE, -1);
+
+          switch (state) {
+          case BluetoothHeadset.STATE_CONNECTED:
+            Log.d(LOG_TAG, "Headset is plugged");
+            mCachedWebView.sendJavascript(
+                "cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetAdded();");
```

Comment:
This is the same javascript and log message as above. Should this be moved into its own method to cut down on the duplication?